config/v3_3_experimental/types/karg.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/shared/errors"
+
+	"github.com/coreos/vcontext/path"
+	"github.com/coreos/vcontext/report"
+)
+
+func (k KernelArgument) Validate(c path.ContextPath) (r report.Report) {
```

Comment:
what should happen when current `cmdline` contains some `foo=boo`and that `karg` is blacklisted with `ShoulndtExist` ?
shouldn't we append `/proc/cmdline` to `ShouldExist` before checking?

Reply:
I'm not sure I follow. This is validating that the config isn't contrasting by saying that a specific kernel argument should and should not exist.

Reply:
ok, i misunderstood here - missed that this check happens before modification of `cmdline`.